### PR TITLE
Node controller change to use field selector

### PR DESCRIPTION
**Pull request by ravigadde**
List the pods specific to a host instead of all.

### CODE REVIEW

**Comment by ravigadde** [comment]
The field selector is version dependent. I suppose spec.host will work, but need to make it more generic. The version is not available in client interface in node controller. Will make additional changes and resubmit the patch.

**Comment by ravigadde** [comment]
@davidopp @ddysher 
Can you please look at this request? Its an optimization in nodecontroller to not pull all the pods when a node is unhealthy. I have no idea what the Shippable failure is.

File: pkg/cloudprovider/nodecontroller/nodecontroller.go [hunk 1]
```diff
@@ -648,15 +648,25 @@ func (nc *NodeController) getCloudNodesWithSpec() (*api.NodeList, error) {
...
+	}
+}
+
 // deletePods will delete all pods from master running on given node.
 func (nc *NodeController) deletePods(nodeID string) error {
 	glog.V(2).Infof("Delete all pods from %v", nodeID)
-	// TODO: We don't yet have field selectors from client, see issue #1362.
-	pods, err := nc.kubeClient.Pods(api.NamespaceAll).List(labels.Everything(), fields.Everything())
+	pods, err := nc.kubeClient.Pods(api.NamespaceAll).List(labels.Everything(),
+		fields.OneTermEqualSelector(getHostFieldLabel(nc.kubeClient.APIVersion()), nodeID))
```
Comment (wojtek-t) [inline pkg/cloudprovider/nodecontroller/nodecontroller.go]: There is not need to introduce the APIVersion mechanism.
Basically, what you should use is "client.PodHost" - see e.g.:
https://github.com/GoogleCloudPlatform/kubernetes/blob/d0288f714328e82f54f782bfa610cb3cb285c17f/pkg/kubelet/config/apiserver.go#L30
And this should be handled correctly by our infrastructure - see:
https://github.com/GoogleCloudPlatform/kubernetes/blob/d0288f714328e82f54f782bfa610cb3cb285c17f/pkg/client/request.go#L295

**Comment by ravigadde** [comment]
@wojtek-t 

Thanks for the comment, updated the pull request with the change.

**Comment by wojtek-t** [comment]
LGTM - will merge on green

@davidopp I hope you will be fine if I review it, although you're assigned to it :)

**Pull request merged**